sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -49,6 +49,7 @@ import org.apache.spark.sql.types._
  *
  */
 abstract class Expression extends TreeNode[Expression] {
+  protected def nullabilitySuffix: String = if (nullable) "<?>" else "<!>"
```

Comment:
This is used to show nullability in the query plan / expression tree. I found it useful while debugging nullability issues.